File: js/gameOver.js
// GameOver.java port

import { GameObject } from './gameObject.js';
import { HCENTER, TOP } from './renderer.js';
import { t } from './i18n.js';

export class GameOver extends GameObject {
    constructor(game, title) {
        super(game);
        this.title = title;
        this.flag = false;
    }

    update() {
        if (this.game.input.isFiring()) {
            this.flag = true;
        } else if (this.flag) {
            this.game.initGame();
        }
    }

    paint(g) {
        const W = this.game.WIDTH;

        g.setColor(0x000000);
        g.fillRect(0, 0, this.game.WIDTH, this.game.HEIGHT);

        g.setColor(0xFFFFFF);
        g.drawString(this.title, W / 2, 20, HCENTER | TOP);
        g.drawString(t('score') + this.game.score, W / 2, 50, HCENTER | TOP);
        g.drawString(t('highScore') + this.game.highScore, W / 2, 70, HCENTER | TOP);
    }
}

File: orig/Game/src/GameOver.java
import javax.microedition.lcdui.*;

public class GameOver extends GameObject
{
    String title;

    public GameOver(String title) {
	this.title = title;
    }

    private boolean flag = false;

    public void update() {
	if (canvas.is_fireing()) {
	    flag = true;
	} else if (flag) {
	    game.initGame();
	}
    }

    public void paint(Graphics g) {
        g.setColor(0x000000);
        g.fillRect(0, 0, canvas.getWidth(), canvas.getHeight());

        g.setColor(0xFFFFFF);
	g.drawString(title,
		     canvas.getWidth()/2, 20,
		     g.HCENTER | g.TOP);

	g.drawString("Score: " + game.score,
		     canvas.getWidth()/2, 50,
		     g.HCENTER | g.TOP);
	g.drawString("High Score: " + game.high_score,
		     canvas.getWidth()/2, 70,
		     g.HCENTER | g.TOP);
    }
}
